scripts/get_mediatype.lua

```diff
@@ -47,31 +47,75 @@ function get_mediatype(mimetype)
 
         return IMAGE
 
-    -- TODO: implement video and audio
+        -- TODO: implement video and audio
 
     else
 
         -- no supported mediatype could be determined
         return false
-
     end
-
-
 end
 
 -------------------------------------------------------------------------------
 -- This function is called from the route to check the file extension of the given filename.
 -- Parameters:
---     'mimetype' (string):  the mimetype of the file.
---     `filename` (string): the name of the file excluding the file extension.
+-- 'mimetype' (string):  the mimetype of the file.
+-- `filename` (string): the name of the file excluding the file extension.
 --
 -- Returns:
---    a boolean indicating whether the file extension is correct or not.
+-- a boolean indicating whether the file extension is correct or not.
 -------------------------------------------------------------------------------
 function check_file_extension(mimetype, filename)
 
     if (mimetype == XML) then
-        return string.sub(filename, -4) == ".xml"
+        local ext = string.sub(filename, -4)
+
+        -- valid extensions are: xml, xsl (XSLT), and .xsd (XML Schema)
+        return ext == ".xml" or ext == ".xsl" or ext == ".xsd"
     end
+end
+
+
+-------------------------------------------------------------------------------
+-- This function splits a string containing a mimetype and a charset into a pair.
+-- The information was sent by the client.
+-- Parameters:
+-- 'mimetype_and_charset' (string): the mimetype and charset of a file, e.g. "application/xml; charset=UTF-8"
+--
+-- Returns:
+-- a pair containing (first) the mimetype, and (second) the charset or false if no charset is given
+function split_mimetype_and_charset(mimetype_and_charset)
 
-end
\ No newline at end of file
+    -- check if mimetype_and_charset contains a semicolon followed by a whitespace character
+    -- mimetype_and_charset might look like this: "application/xml; charset=UTF-8"
+    -- string.find returns 'nil' if the pattern was not found
+    local start_char, end_char = string.find(mimetype_and_charset, "; ")
+
+    -- make sure string.find did not return 'nil' and both start and end characters have been returned
+    if start_char ~= nil and (start_char + 1) == end_char then
+        -- pattern "; " was found in string
+
+        -- original mimetype also contains the encoding, e.g. "application/xml; charset=UTF-8"
+        -- split the string using the position information start_char and end_char
+
+        -- get a substring from the beginning until the position of "; "
+        -- substract 1 from start_char because Lua includes the last character which we do not want
+        local mimetype = string.sub(mimetype_and_charset, 1, start_char -1) -- e.g. "application/xml"
+
+        -- get a substring from the end of "; " until the last character
+        -- add 1 to end_char because otherwise Lua includes the whitespace which we do not want
```

Comment:
The whitespace after the semicolon is optional: https://tools.ietf.org/html/rfc7231#section-3.1.1